bsdfdisk/create_part.cgi

```diff
@@ -29,15 +29,18 @@ $part->{'letter'} = $in{'letter'};
 $in{'start'} =~ /^\d+$/ || &error($text{'nslice_estart'});
 $in{'end'} =~ /^\d+$/ || &error($text{'nslice_eend'});
 $in{'start'} < $in{'end'} || &error($text{'npart_erange'});
-$part->{'startblock'} = $in{'start'};
+$part->{'startblock'} = int($in{'start'}/2048)*2048;  # 1MB alignment
 $part->{'blocks'} = $in{'end'} - $in{'start'};
 
 # Slice type
 $part->{'type'} = $in{'type'};
+# Set partition properties
+$part->{'label'} = $in{'label'} if ($in{'label'} =~ /^[a-zA-Z0-9._-]+$/);
```

Comment:
@karmantyu, I have also changed the regex from `/^[a-zA-Z0-9._-]*$/` to `/^[a-zA-Z0-9._-]+$/` because we expect the label not to be empty.